### PR TITLE
Capture Client Guest OS and architecture in JDBC

### PR DESCRIPTION
**Pull request by muskan124947**
Use case: Update the application name to "Microsoft JDBC - {OS}, {Platform} - {architecture}"

### CODE REVIEW

**Review by machavan — COMMENTED** [review]
We need to set the appname in the format 

Microsoft JDBC - {OS}, {Platform} - {architecture}

in the enum SQLServerDriverStringProperty .

Currently it is set to default value 


    APPLICATION_NAME("applicationName", SQLServerDriver.DEFAULT_APP_NAME),

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDriver.java [hunk 1]
```diff
@@ -1251,6 +1266,7 @@ public java.sql.Connection connect(String url, Properties suppliedProperties) th
                     "Microsoft JDBC Driver " + SQLJdbcVersion.MAJOR + "." + SQLJdbcVersion.MINOR + "."
                             + SQLJdbcVersion.PATCH + "." + SQLJdbcVersion.BUILD + SQLJdbcVersion.RELEASE_EXT
                             + " for SQL Server");
+            logClientOSAndArchInfo();
```
Comment (lilgreenbird) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDriver.java:1269]: is this meant to be used elsewhere? if only used once here then there's no need to create a sep method

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDriver.java [hunk 2]
```diff
@@ -1240,6 +1240,21 @@ static String getPropertyOnlyName(String name, Logger logger) {
...
+     * and logs this information.
+     */
+    private void logClientOSAndArchInfo() {
+        try {
+            String osName = System.getProperty("os.name");
+            String osArch = System.getProperty("os.arch");
+            
+            loggerExternal.log(Level.FINE, "Client OS: " + osName + ", Architecture: " + osArch);
+        } catch (Exception e) {
+            loggerExternal.warning("Unable to capture client OS and architecture information.");
```
Comment (lilgreenbird) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDriver.java:1255]: should add e.getMessage() for more info

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDriver.java [hunk 3]
```diff
@@ -731,7 +731,33 @@ public final class SQLServerDriver implements java.sql.Driver {
...
+        String javaVmName = System.getProperty("java.vm.name", "");
+        String javaVmVersion = System.getProperty("java.vm.version", "");
+
+        String platform = javaVmName.isEmpty() || javaVmVersion.isEmpty() ? "" : javaVmName + " " + javaVmVersion;
+
+        String appName = String.format(APP_NAME_TEMPLATE, osName, platform, osArch);
+        
+        // If all values are empty, return the default app name
+        if (osName.isEmpty() && platform.isEmpty() && osArch.isEmpty()) {
+            return DEFAULT_APP_NAME;
```
Comment (divang) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDriver.java:755]: If osName, platform and osArch are empty, then we can avoid String formatting: 
   
  if (!osName.isEmpty() && !platform.isEmpty() && !osArch.isEmpty()) {
      return  String.format(APP_NAME_TEMPLATE, osName, platform, osArch); 
  }
  return DEFAULT_APP_NAME;

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDriver.java [hunk 4]
```diff
@@ -1028,6 +1054,10 @@ String getClassNameLogging() {
                 drLogger.finer("Error registering driver: " + e);
             }
         }
+        if (loggerExternal.isLoggable(Level.INFO)) {
```
Comment (divang) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDriver.java:1057]: For consistent logging level, please use Level.FINE

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDriver.java [hunk 5]
```diff
@@ -563,7 +563,7 @@ static PrepareMethod valueOfString(String value) throws SQLServerException {
 
 enum SQLServerDriverStringProperty {
     APPLICATION_INTENT("applicationIntent", ApplicationIntent.READ_WRITE.toString()),
-    APPLICATION_NAME("applicationName", SQLServerDriver.DEFAULT_APP_NAME),
```
Comment (machavan) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDriver.java:566]: Can we keep this unchanged to default to DEFAULT_APP_NAME

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDriver.java [hunk 6]
```diff
@@ -741,7 +762,7 @@ public final class SQLServerDriver implements java.sql.Driver {
                     SQLServerDriverStringProperty.APPLICATION_INTENT.getDefaultValue(), false,
                     new String[] {ApplicationIntent.READ_ONLY.toString(), ApplicationIntent.READ_WRITE.toString()}),
             new SQLServerDriverPropertyInfo(SQLServerDriverStringProperty.APPLICATION_NAME.toString(),
-                    SQLServerDriverStringProperty.APPLICATION_NAME.getDefaultValue(), false, null),
+                    "", false, null),
```
Comment (machavan) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDriver.java:765]: Can we keep this unchanged to default to DEFAULT_APP_NAME

File: src/test/java/com/microsoft/sqlserver/jdbc/SQLServerDriverTest.java [hunk 7]
```diff
@@ -190,4 +190,22 @@ public void testConnectionDriver() throws SQLException {
...
+    /**
+     * test application name
+     * 
+     * @throws SQLException
+     */
+    @Test
+    public void testApplicationName() throws SQLException {
+        try (Connection conn = DriverManager.getConnection(connectionString);
+             Statement stmt = conn.createStatement();
+             ResultSet rs = stmt.executeQuery("SELECT program_name FROM sys.dm_exec_sessions WHERE session_id = @@SPID")) {
```
Comment (machavan) [inline src/test/java/com/microsoft/sqlserver/jdbc/SQLServerDriverTest.java:203]: Good to have this test which is actually querying the program name

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDriver.java (hunk 5 above)
Comment (muskan124947) [reply src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDriver.java]: Sure, Its not needed

File: src/test/java/com/microsoft/sqlserver/jdbc/SQLServerDriverTest.java (hunk 7 above)
Comment (muskan124947) [reply src/test/java/com/microsoft/sqlserver/jdbc/SQLServerDriverTest.java:203]: Yes

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDriver.java [hunk 8]
```diff
@@ -731,7 +731,28 @@ public final class SQLServerDriver implements java.sql.Driver {
...
+     * If any of the properties cannot be fetched, it falls back to the default application name.
+     * Format -> Microsoft JDBC - {OS}, {Platform} - {architecture}
+     *
+     * @return the constructed application name or the default application name if properties are not available
+     */
+    static String getAppName() {
+        String osName = System.getProperty("os.name", "");
+        String osArch = System.getProperty("os.arch", "");
+        String javaVmName = System.getProperty("java.vm.name", "");
+        String javaVmVersion = System.getProperty("java.vm.version", "");
```
Comment (lilgreenbird) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDriver.java:747]: javaVmName and javaVmVersion are not used?

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDriver.java [hunk 9]
```diff
@@ -731,7 +731,28 @@ public final class SQLServerDriver implements java.sql.Driver {
     static final String AUTH_DLL_NAME = "mssql-jdbc_auth-" + SQLJdbcVersion.MAJOR + "." + SQLJdbcVersion.MINOR + "."
             + SQLJdbcVersion.PATCH + "." + Util.getJVMArchOnWindows() + SQLJdbcVersion.RELEASE_EXT;
     static final String DEFAULT_APP_NAME = "Microsoft JDBC Driver for SQL Server";
+    static final String APP_NAME_TEMPLATE = "Microsoft JDBC - %s, %s - %s";
```
Comment (lilgreenbird) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDriver.java:734]: should this have "Driver" and "SQL Server" in there? old name is "Microsoft JDBC Driver for SQL Server" new name will just be something like "Microsoft JDBC - Windows 11, 17.0.1+12-39- amd64"

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDriver.java (hunk 3 above)
Comment (lilgreenbird) [reply src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDriver.java:751]: this should be a static and only done once

**Review by Jeffery-Wasty — APPROVED** [review]

**Review by divang — APPROVED** [review]

**Pull request merged**